LibreNMS/Snmptrap/Handlers/AdvaNetThresholdCrossingAlert.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * AdvaNetThresholdCrossingAlert.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Adva Threshold Exceeded Alarms.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2018 KanREN, Inc
+ * @author     Heath Barnhart <hbarnhart@kanren.net> & Neil Kahle <nkahle@kanren.net>
+ */
+
+namespace LibreNMS\Snmptrap\Handlers;
+
+use App\Models\Device;
+use LibreNMS\Interfaces\SnmptrapHandler;
+use LibreNMS\Snmptrap\Trap;
+use Log;
+
+class AdvaNetThresholdCrossingAlert implements SnmptrapHandler
+{
+    /**
+     * Handle snmptrap.
+     * Data is pre-parsed and delivered as a Trap.
+     *
+     * @param Device $device
+     * @param Trap $trap
+     * @return void
+     */
+    public function handle(Device $device, Trap $trap)
+    {
+        $interval = $trap->getOidData($trap->findOid("CM-PERFORMANCE-MIB::cmEthernetNetPortThresholdInterval"));
+        $ifName = $trap->getOidData($trap->findOid("IF-MIB::ifName"));
+        $threshMessage = $this->getThresholdMessage(
+            $trap->getOidData($trap->findOid("CM-PERFORMANCE-MIB::cmEthernetNetPortThresholdVariable"))
+        );
+
+        Log::event("$ifName $threshMessage threshold exceeded for $interval", $device->device_id, 'trap', 2);
+    }
+    public function getThresholdMessage($thresholdOid)
+    {
+        foreach ($this->getThresholds() as $oid => $descr) {
+            if (str_contains($thresholdOid, $oid)) {
+                return $descr;
+            }
+        }
+        return 'unknown';
+    }
+    public function getThresholds()
```

Comment:
This seems to be duplicated

Reply:
The function name is the same, but the array of values returned is not. One is for Ethernet access ports and the other is for Ethernet network ports. Really the only difference between the two arrays are three characters in the keys that identify the trap as coming from an access port (cm**Acc**PortQosShaperStatsBT) or a network port (cm**Net**PortQosShaperStatsBT). I think I can refactor it down though.